WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentOnboardingAdaptor.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+import Combine
+import protocol Yosemite.StoresManager
+
+/// This is really a re-implementation of the CardPresentPaymentsOnboardingPresenter, as it needs to take the calls to `showOnboardingIfRequired` and
+/// route the output to a SwiftUI view for display, rather than directly displaying on the viewController that's passed in.
+final class CardPresentPaymentsOnboardingPresenterAdaptor: CardPresentPaymentsOnboardingPresenting {
+    private let stores: StoresManager
+
+    private let onboardingUseCase: CardPresentPaymentsOnboardingUseCase
+
+    private let readinessUseCase: CardPresentPaymentsReadinessUseCase
+
+    private let onboardingViewModel: CardPresentPaymentsOnboardingViewModel
+
+    private var readinessSubscription: AnyCancellable?
+
+    private(set) var onboardingScreenViewModelPublisher: AnyPublisher<CardPresentPaymentOnboardingPresentationEvent, Never>
+
+    private let onboardingScreenViewModelSubject: PassthroughSubject<CardPresentPaymentOnboardingPresentationEvent, Never> = PassthroughSubject()
+
+    init(stores: StoresManager = ServiceLocator.stores) {
+        self.stores = stores
```

Comment:
nit: does `stores` need to be held by the adaptor? from the implementation it looks like it's just passed to `CardPresentPaymentsOnboardingUseCase` and `CardPresentPaymentsReadinessUseCase`, but maybe there are other plans to use it in a later task?

Reply:
No, you're right, I'll remove the property and we can add it if we need it in future